definitions/push-application-snapshot/push-application-snapshot.yaml

```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: push-application-snapshot-task
+spec:
+  params:
+    - name: mappedApplicationSnapshot
+      description: |
+        JSON string representing the ApplicationSnapshot
+        For example:
+        """
+        {
+          "application": "foo",
+          "images": [
+            {
+              "component": "component_fizz_1",
+              "pullspec": "quay.io/src_namespace/src_repo_name@sha256:digest",
+              "targetRepo": "quay.io/dest_namespace/dest_repo_name"
+            },
+            {
+              "component": "component_bang_3",
+              "pullspec": "quay.io/src_namespace/src_repo_name",
+              "targetRepo": "quay.io/dest_namespace/dest_repo_name"
+            }
+          ]
+        }
+        """
+        Note:
+        * The pullspec digest (e.g. @sha256:...) may be ommited.
+        * If the pullspec digest is ommited, then assume "latest".
+      type: string
+    - name: retries
+      description: Retry copy N times.
+      type: int
+      default: 0
+    - name: destinationTag
+      description: option tag to give destination images.
+      type: string
+      default: ""
+  steps:
+    - name: mappedApplicationSnapshot
+      image: http://quay.io/hacbs-release/release-utils:latest
+      script: |
+        if [[ -z "$(params.mappedApplicationSnapshot)" ]]
+        then
+          exit 1 # Missing URL parameters
+        fi
+        app=$(jq -r '.application' <<<$(params.mappedApplicationSnapshot))
+        tag="$(params.destinationTag)"
+        printf 'Begining push task for application: "%s"\n' "$app"
+        while read line;
+        do
+          set -- $(jq -cr '.[]' <<<"$line" | xargs)
```

Comment:
If I got it correctly, you are setting here new params and expecting `$1` to be the `pullspec` and `$2` to be the `repository`. Thing is that json have no order, so this is a bit dangerous. I prefer you to set variables by quering the json itself.

```
pullspec=$(jq -rc '.pullspec' <<< "$line" | xargs)
repository=$(jq -rc '.repository' <<< "$line" | xargs)
```
I get that you want to be agnostic to field name changes, but given the order issue I think we have to go for this.